dbmongo/js/compact/currentState.ts

```diff
@@ -16,17 +16,18 @@ export function currentState(batches: BatchValue[]): CurrentDataState {
         }
       }
 
+      type ValueOf<T> = T[keyof T]
+      const objectEntries = <T>(obj: T): [keyof T, ValueOf<T>][] =>
+        (Object.keys(obj) as (keyof T)[]).map((key) => [key, obj[key]])
+
       //2. On ajoute les nouvelles clés
-      const neyKeyTypes = Object.keys(batch).filter(
-        (type) => type !== "compact"
-      ) as (keyof BatchValue)[]
-      neyKeyTypes.forEach((type) => {
+      for (const [type] of objectEntries(batch)) {
```

Comment:
Je ne comprends pas: on récupère la première valeure de `objectEntries(batch)` ?
Dans ce cas, où utilise-t-on la seconde ?

Reply:
Oui, ici `[type]` est une *déconstruction* partielle de chaque élément (de type `[key, value]`) itéré par `for` depuis le tableau retourné par `objectEntries(batch)`. On ne récupère que le premier élément (`key`) et ignore la `value` car on peut la retrouver par l'expression `batch[type]` employée plus bas, que je trouvais plus lisible et explicite que l'introduction d'une n-ième variable. Ceci dit, ce n'est qu'une préférence personnelle.

Reply:
Pour info, si on avait voulu récupérer les deux, on aurait écrit `for (const [type, value] of objectEntries(batch)) {`, ce qui nous aurait donné le droit d'employer `value` au lieu de `batch[type]`, plus bas.

Reply:
Pour référence, Pierre suggère la définition d'une fonction `typedObjectKeys` au lieu de `objectEntries`, pour améliorer la lisibilité, en supposant que les clés seules sont utilisées plus souvent que les couples clé / valeur. 👍

Reply:
c'est fait !